temporian/core/operators/lag.py

```diff
@@ -84,21 +86,67 @@ def build_op_definition(cls) -> pb.OperatorDef:
 operator_lib.register_operator(LagOperator)
 
 
-def lag(event: Event, duration: Duration) -> Event:
-    if duration <= 0:
-        raise ValueError("duration must greater than zero")
+def _base_lag(
```

Comment:
Why creating an implementation function here? It seems this is the entirety of the